### PR TITLE
Update to crystal 0.24

### PR DESCRIPTION
**Pull request by bcardiff**
There are some breaking changes and known issues with 0.24.1 that should be fixed by the following changes.

### CODE REVIEW

**Comment by bcardiff** [comment]
I just noticed #2. In this PR there some additional chances done since the new method already filter . and .. 

Besides updating the shard.yml to clarify the expected crystal version.

File: shard.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ authors:
 
 license: MIT
 
-crystal: 0.23.1
+crystal: 0.24.0
```
Comment (drujensen) [inline shard.yml:9]: should this be 0.24.1?

Comment (bcardiff) [reply shard.yml:9]: The breaking change was in 0.24.0 but it’s almost the same to use .0 or .1 since 0.24.0 could be treated as a yanked release.

**Review by drujensen — APPROVED** [review]

File: shard.yml (hunk 1 above)
Comment (drujensen) [reply shard.yml:9]: @bcardiff @mosop I don't have permission to merge this

**Comment by mosop** [comment]
Dir.each was re-added in Crystal 0.24.2. So I don't support the earlier versions for simplicity. Thank you all!

**Pull request closed without merging**